src/JNI/com/amazonaws/kinesis/video/producer/jni/Parameters.cpp

```diff
@@ -1188,3 +1193,131 @@ BOOL releaseStreamDataBuffer(JNIEnv* env, jobject dataBuffer, UINT32 offset, PBY
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    jmethodID methodId = NULL;
+    CHECK(env != NULL && streamEventMetadata != NULL && pStreamEventMetadata != NULL);
+    const char *retChars;
+
+    // Load KinesisVideoFrame
+    jclass cls = env->GetObjectClass(streamEventMetadata);
+    if (cls == NULL) {
+        DLOGE("Failed to create StreamEventMetadata class.");
+        CHK(FALSE, STATUS_INVALID_OPERATION);
+    }
+
+    // Retrieve the methods and call them
+
+    methodId = env->GetMethodID(cls, "getVersion", "()I");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getVersion");
+    } else {
+        pStreamEventMetadata->version = env->CallIntMethod(streamEventMetadata, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+
+    methodId = env->GetMethodID(cls, "getImagePrefix", "()Ljava/lang/String;");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getImagePrefix");
+    } else {
+        jstring retString = (jstring) env->CallObjectMethod(streamEventMetadata, methodId);
+        CHK_JVM_EXCEPTION(env);
+
+        if (retString != NULL) {
+            jsize javaStringLength = env->GetStringUTFLength(retString);
+            CHK(javaStringLength <= MAX_IMAGE_PREFIX_LENGTH, STATUS_INVALID_ARG_LEN);
+            pStreamEventMetadata->imagePrefix = (PCHAR)MEMCALLOC(javaStringLength + 1, SIZEOF(CHAR));
+            retChars = env->GetStringUTFChars(retString, NULL);
+            STRNCPY(pStreamEventMetadata->imagePrefix, retChars, javaStringLength);
+            env->ReleaseStringUTFChars(retString, retChars);
+        }
+    }
+
+    methodId = env->GetMethodID(cls, "getNumberOfPairs", "()B");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getNumberOfPairs");
+    } else {
+        pStreamEventMetadata->numberOfPairs = env->CallByteMethod(streamEventMetadata, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+
+    methodId = env->GetMethodID(cls, "getNames", "()[Ljava/lang/String;");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getNames");
+    } else {
+        if(!allocStreamEventMetadataArray(env, streamEventMetadata, pStreamEventMetadata->names, methodId)) {
+            DLOGW("Failed in allocStreamEventMetadataArray step for metadata names.");
+        }
+    }
+
+    methodId = env->GetMethodID(cls, "getValues", "()[Ljava/lang/String;");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getValues");
+    } else {
+        if(!allocStreamEventMetadataArray(env, streamEventMetadata, pStreamEventMetadata->values, methodId)) {
+            DLOGW("Failed in allocStreamEventMetadataArray step for metadata vlaues.");
+        }
+    }
+
+CleanUp:
+    return STATUS_FAILED(retStatus) ? FALSE : TRUE;
+}
+
+BOOL allocStreamEventMetadataArray(JNIEnv* env, jobject streamEventMetadata, PCHAR* metaDataArray, jmethodID methodId) {
+    STATUS retStatus = STATUS_SUCCESS;
+    jobjectArray retArray = NULL;
+    jsize namesArrayLength = 0;
```

Comment:
Recommend renaming this to something generic as this function is used for name and value both

Reply:
Yes nice catch!